pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -302,6 +305,9 @@ func (dswp *desiredStateOfWorldPopulator) processPodVolumes(pod *v1.Pod) {
 	// some of the volume additions may have failed, should not mark this pod as fully processed
 	if allVolumesAdded {
 		dswp.markPodProcessed(uniquePodName)
+		// New pod has been synced. Re-mount all volumes that need it
+		// (e.g. DownwardAPI)
+		dswp.actualStateOfWorld.MarkRemountRequired(uniquePodName)
```

Comment:
this commit showed up in a bisect of scalability test regression of pod startup time (https://github.com/kubernetes/kubernetes/issues/60589#issuecomment-372485732), and this change looks odd... does this force double mount setup for all pods?

Reply:
The flow that leads to `MarkRemountRequired` call is:

- `kubelet.syncPod` is called (on pod update or every 90s)
  -  `volumeManager.WaitForAttachAndMount` is called and marks the pod for reprocessing

- `desiredStateOfWorldPopulator.findAndAddNewPods` is called periodically
    - `desiredStateOfWorldPopulator.processPodVolumes` is called (only when the pod was marked for reprocessing in `syncPod`)
        - `actualStateOfWorld.MarkRemountRequired` is called
            - reconciler re-mounts the volumes (updates secrets, downward API, ...). Update of DownwardAPI is the reason why `syncPod` is involved.

Notice that before this PR, `MarkRemountRequired` was called directly by `kubelet.syncPod`, so the frequency of the calls did not change. With this PR, they are called only in the right order. I don't think it could affect pod startup time.

On the other hand, `syncPod` is called with every pod update and pod changes quite often during startup. It is possible that I missed something there.